service/src/main.rs

```diff
@@ -376,6 +377,16 @@ fn start_worker<E, T, D, InitializationHandler>(
 	enclave
 		.set_nonce(nonce)
 		.expect("Could not set nonce of enclave. Returning here...");
+
+	let metadata = node_api.clone().metadata;
```

Comment:
Why do we clone the `node_api` here? Shouldn't we clone the metadata itself instead?